Sources/Container.swift

```diff
@@ -224,6 +238,10 @@ extension Container: _Resolver {
             var resolvedInstance: Service?
             let key = ServiceKey(serviceType: Service.self, argumentsType: Arguments.self, name: name, option: option)
 
+            if key == Self.graphIdentifierKey {
+                return currentObjectGraph as! Service?
```

Comment:
We will crash if this cast fails.
Would it make sense to throw an error with a message instead? Or default to a nil Service?

Reply:
Hm, when would this happen? The cast is always to `GraphIdentifier?` as `graphIdentifierKey` ensures we're making a call to resolve a GraphIdentifier.

If the `currentObjectGraph` is value `nil` that will successfully cast to `GraphIdentifier?`.